include/odp/api/spec/event.h

```diff
@@ -149,6 +149,28 @@ uint64_t odp_event_to_u64(odp_event_t hdl);
  */
 void odp_event_free(odp_event_t event);
 
+/**
+ * Free multiple events
+ *
+ * Otherwise like odp_event_free(), but frees multiple events to their
+ * originating pools.
+ *
+ * @param event    Array of event handles
+ * @param num      Number of events to free
+ */
+void odp_event_free_multi(const odp_event_t event[], int num);
+
+/**
+ * Free multiple events to the same pool
+ *
+ * Otherwise like odp_event_free_multi(), but all events must be from the
+ * same originating pool.
+ *
+ * @param event    Array of event handles
+ * @param num      Number of events to free
+ */
+void odp_event_free_sp(const odp_event_t event[], int num);
```

Comment:
Not clear why this is needed. How does an application determine this more efficiently than the implementation would?

Reply:
Application knows the configuration, so it knows when all packets (e.g. from certain queue/pkt input/etc) are allocated from the same pool. When that's the case, free implementation can be more efficient (than today) as it does not have to access/check/sort all packets of the burst. It can just free all into the pool of the first packet.